rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/ReSubmitJobCommand.java

```diff
@@ -0,0 +1,169 @@
+    public ReSubmitJobCommand(String... params) throws NullPointerException {
+        Objects.requireNonNull(params);
+
+        this.jobId = params[0];
+        if (params.length > 1) {
+            this.variables = params[1];
+        }
+        this.jobKeyValueTransformer = new JobKeyValueTransformer();
+
+    }
+
+    @Override
+    public void execute(ApplicationContext currentContext) throws CLIException {
+
+        try {
+            validateFilePath(currentContext);
+            File jobFile = new File(pathname);
+            String contentType = URLConnection.getFileNameMap().getContentTypeFor(pathname);
+            JobIdData jobId;
+            try (FileInputStream inputStream = new FileInputStream(jobFile)) {
+                if (APPLICATION_XML.getMimeType().equals(contentType)) {
+                    jobId = currentContext.getRestClient().submitXml(currentContext.getSessionId(),
+                                                                     inputStream,
+                                                                     map(this.variables));
+                } else {
+                    jobId = currentContext.getRestClient().submitJobArchive(currentContext.getSessionId(),
+                                                                            inputStream,
+                                                                            map(this.variables));
+                }
+            }
+            writeLine(currentContext, "Job('%s') successfully submitted: job('%d')", pathname, jobId.getId());
+            resultStack(currentContext).push(jobId);
+        } catch (Exception e) {
+            handleError(String.format("An error occurred while attempting to submit job('%s'):", pathname),
+                        e,
+                        currentContext);
+        }
+
+    }
+
+    private void validateFilePath(ApplicationContext currentContext) throws JobCreationRestException {
+
+        if (!isFilePathValid(pathname)) {
+            throw new CLIException(REASON_INVALID_ARGUMENTS, String.format("'%s' is not a valid file.", pathname));
+        }
+
+        if (!isValidFileMimeType(pathname)) {
+            throw new JobCreationRestException("Unknown job descriptor type: " + pathname);
+        }
+
+        if (isFileExisting(pathname) && isFileEmpty(pathname)) {
+            throw new CLIException(REASON_FILE_EMPTY, String.format("'%s' is empty.", pathname));
+        }
+
+        if (!isFileExisting(pathname)) {
+            throw new CLIException(REASON_INVALID_ARGUMENTS, String.format("'%s' does not exist.", pathname));
+        }
+
+    }
+
+    private Map<String, String> map(String variables) {
+        return jobKeyValueTransformer.transformVariablesToMap(variables);
+    }
+
+    private Boolean isFileEmpty(String pathname) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(pathname))) {
+            if (isFileExisting(pathname)) {
+                if (reader.readLine() == null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use or store the value returned from "readLine" instead of throwing it away. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2677)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)